src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -142,9 +272,17 @@ protected override void Dispose(bool disposing)
                 if (_pinNumberToSafeLineHandle.TryGetValue(pin, out SafeLineHandle pinHandle))
                 {
                     Interop.ReleaseGpiodLine(pinHandle);
+                    _pinNumberToSafeLineHandle.Remove(pin);
                 }
+            }
 
-                _pinNumberToSafeLineHandle.Remove(pin);
+            foreach (int pin in _pinNumberToEventHandler.Keys)
+            {
+                if (_pinNumberToEventHandler.TryRemove(pin, out LibGpiodDriverEventHandler pinEventHandler)) {
```

Comment:
you should have a function for those few lines of code since you have them in at least 2 places